app/models/efile_submission_transition.rb

```diff
@@ -22,25 +22,22 @@
 #
 class EfileSubmissionTransition < ApplicationRecord
   belongs_to :efile_submission, inverse_of: :efile_submission_transitions, touch: true
+  has_many :efile_submission_transition_errors
+  has_many :efile_errors, through: :efile_submission_transition_errors
 
   after_destroy :update_most_recent, if: :most_recent?
+  after_create :persist_efile_error_from_metadata
```

Comment:
Consider hooking [after_create_commit](https://apidock.com/rails/v6.0.0/ActiveRecord/Transactions/ClassMethods/after_create_commit) to be sure the transition is saved to the database before this parsing code runs.

Or a unit test that covers that scenario.

Reply:
👍 changed